.github/workflows/ci_workflows.yml

```diff
@@ -39,3 +39,37 @@ jobs:
       run: make check
     - name: make distcheck
       run: make distcheck
+
+  tests-meson:
+    name: ${{ matrix.name }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: linux (with gcc and meson)
+            os: ubuntu-latest
+
+          - name: MacOS (with clang and meson)
+            os: macos-latest
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Install dependencies
+      run: |
+        if [ "$RUNNER_OS" = Linux ]; then
+            sudo apt-get -y install meson ninja-build
+        else
+            brew install meson
+        fi
+
+    - name: configure
+      run: meson setup builddir --fatal-meson-warnings
+    - name: build
+      run: ninja -C builddir
```

Comment:
Maybe use the `meson` equivalent?

Reply:
Obviously, it really doesn't matter though I find it easier to just use the one tool and let it do what it wants under the hood, so, `meson compile -C builddir` -- this is what the documentation suggests too: https://mesonbuild.com/Running-Meson.html

Reply:
The documentation simply states both are possible.

`meson compile` exists basically for one reason: if you don't know or care what build backend you used, it will detect whether to run ninja or msbuild or xcodebuild for you.

To do that, it imports 70k lines of code with the only active component being `subprocess.run([detect_ninja(), '-c', build_dir])`

Reply:
OK, happy to stay with what you have!